Add Support for TaxAreaLookup

rypit opened the pull request:
In order to improve performance with quotation and invoice we have been advised to include `tax_area_id` in customer payloads. This should prevent a "double hop" scenario since we will not need to geocode addresses to resolve the id at the time of request.

In order to get the tax area ids, we will add a new endpoint and allow client applications to geocode addresses ahead of time, such as when payments are initially collected.

drinks reviewed (approved):
This looks nice and clean! Thanks for getting this put together so quickly.

rypit commented on lib/vertex_client/configuration.rb:
```diff
@@ -12,12 +12,13 @@ class Configuration
     def initialize
       @trusted_id = ENV['VERTEX_TRUSTED_ID']
-      @soap_api   = ENV['VERTEX_SOAP_API']
+      @soap_api   = VERTEX_SOAP_API
```
Yeah, but we will address with our initializer instead.

rypit commented on lib/vertex_client/connection.rb:
```diff
@@ -30,6 +30,7 @@ def transform_payload
     # TODO Consider removing this conditional to make this more robust
     def handle_response(response)
       if response
+        return TaxAreaResponse.new(response, @payload.response_key) if @payload.is_a? TaxAreaPayload
```
Thinking that long term we may want to have Payloads able to handle their own responses as a way to clean up this nasty bit.

rypit committed:
Support varying endpoints

In preparation for adding the TaxAreaLookup call, we need to allow the client to hit varying endpoints.

Note: We will need to update configuration to reflect this.

rypit committed:
Support TaxAreaLookup endpoint

Also allows tax_area_id to be passed to quotation and invoice calls for performance gains.

rypit committed:
Bump to version 0.4.0

Add Tax Lookup & Update VERTEX_SOAP_API

This configuration should be the base of the integration point rather than the specific CalculateTax70 endpoint.

rypit committed:
Allow tax_area_lookup to return a FallbackResponse

rypit commented on lib/vertex_client/fallback_response.rb:
```diff
@@ -4,6 +4,9 @@ class FallbackResponse
 
+    # Temporary hack for tax_area_lookup
+    attr_reader :tax_area_id
+
```
Now that we have a new endpoint in the mix, a lot of our previous abstractions are now broken. We will need to follow up with a refactor to get this codebase into good shape.
